sources/packages/api/src/route-controllers/assessment/assessment.controller.service.ts

```diff
@@ -79,4 +90,56 @@ export class AssessmentControllerService {
     });
     return flattenedAppealRequests;
   }
+
+  /**
+   * Get the notice of assessment data from the assessment.
+   * @param assessmentId assessment id to be retrieved.
+   * @param userId optional user for authorization when needed.
+   * @returns notice of assessment data.
+   */
+  async getAssessmentNOA(
+    assessmentId: number,
+    userId?: number,
+  ): Promise<AssessmentNOAAPIOutDTO> {
+    const assessment = await this.assessmentService.getAssessmentForNOA(
+      assessmentId,
+      userId,
+    );
+
+    if (!assessment) {
+      throw new NotFoundException("Assessment was not found.");
+    }
+
+    if (!assessment.assessmentData) {
+      throw new UnprocessableEntityException(
+        "Notice of assessment data is not present.",
+      );
+    }
+
+    //Disbursement data is populated with dynamic key in a defined pattern to be compatible with form table.
+    const disbursementDetails = {};
+    assessment.disbursementSchedules.forEach((schedule, index) => {
+      const disbursementIdentifier = `disbursement${index + 1}`;
+      disbursementDetails[`${disbursementIdentifier}Date`] = dateString(
```

Comment:
may I know why this logic? for my understanding

Reply:
This is not new, it was just moved, but this dynamically builds the values that are used on the NOA to display the disbursement values as shown below.
![image](https://user-images.githubusercontent.com/61259237/166583798-b7cc2310-55f4-42cf-bec3-4cd284ad1de1.png)